File: paymentsense-coding-challenge-website/src/app/services/country.ts
import { Currency } from './currency';
import { Language } from './Language';

export class Country {
  Alpha2Code: string;
  Name: string;
  Flag: string;
  CallingCodes: string[];
  Capital: string;
  Region: string;
  Population: number;
  Demonym: string;
  Timezones: string[];
  Currencies: Currency[];
  Languages: Language[];
  Borders: string[];
}

File: paymentsense-coding-challenge-api/Paymentsense.Coding.Challenge.Api/Models/Country.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Paymentsense.Coding.Challenge.Api.Models
{
    public class Country
    {
        public string Alpha2Code { get; set; }
        public string Name { get; set; }
        public string Flag { get; set; }
        public string[] CallingCodes { get; set; }
        public string Capital { get; set; }
        public string Region { get; set; }
        public int Population { get; set; }
        public string Demonym { get; set; }
        public string[] Timezones { get; set; }
        public Currency[] Currencies { get; set; }
        public Language[] Languages { get; set; }
        public string[] Borders { get; set; }
    }
}
